File: backend/Congratulatory/Core/DTOs/User/UserResponse.cs
using CoreService.Core.DTOs.UserAccount;

namespace CoreService.Core.DTOs.User;

public class UserResponse
{
    public Guid Id { get; set; }
    public string UserName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public ICollection<UserAccountResponse> UserAccounts { get; set; } = new List<UserAccountResponse>();
}

public class LoginResponse
{
    public string Token { get; set; } = string.Empty;
    public UserResponse User { get; set; } = null!;
}

File: frontend/src/types/user/UserResponse.ts
import type { BirthdayResponse } from '../birthday/BirthdayResponse'
import type { UserAccountResponse } from '../userAccount/UserAccountResponse'

export interface UserResponse {
  id: string
  userName: string
  email: string
  createdAt: string
  updatedAt: string
  birthdays: BirthdayResponse[]
  userAccounts: UserAccountResponse[]
}

export interface LoginResponse {
  token: string
  user: UserResponse
}